src/app/events/[slug]/components/ScheduleTabs.tsx

```diff
@@ -0,0 +1,99 @@
+'use client'
+
+import { useEffect, useRef } from 'react'
+
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '@headlessui/react'
+import theme from 'tailwindcss/defaultTheme'
+
+import type { Event } from '@/types/eventType'
+
+import { BasicCard } from '@/components/BasicCard'
+import { Heading } from '@/components/Heading'
+import { TextLink } from '@/components/TextLink'
+
+import { formatDate, formatTime } from '../utils/dateUtils'
+
+type ScheduleTabsProps = {
+  schedule: Event['schedule']
+}
+
+const { screens } = theme
+
+export function ScheduleTabs({ schedule }: ScheduleTabsProps) {
+  const tabGroupRef = useRef<HTMLDivElement>(null)
+  const hasMounted = useRef(false)
+
+  const validDays = schedule!.days.filter((day) => day.events.length > 0)
```

Comment:
TypeScript think that `schedule` can be `undefined` but, at this point, we know for sure that `schedule` isn't `undefined` since we check for it with `eventHasSchedule` - and only then render the component. 

> You can postfix an expression with ! to tell TypeScript that you know it's not null or undefined. This works the same as an 'as' assertion.

Reply:
I think this is amazing.